src/java.base/share/classes/sun/nio/ch/IOUtil.java

```diff
@@ -514,7 +516,7 @@ static void releaseScopes(Runnable releasers) {
             releasers.run();
     }
 
-    static record LinkedRunnable(Runnable node, Runnable next)
+    record LinkedRunnable(Runnable node, Runnable next)
```

Comment:
Some (optional) style comments: I'm not sure this is a record. E.g. both this and the `Releaser` class are only used externally to call their `run` method. The fact that they are made up of components is immaterial to clients, which seems to suggest that an interface would be better - at least subjectively. If I were to write the code I will declare the implementation inside the factories, as local/anon classes.

Reply:
Separately, also (optional) stylistic: the indenting on this and following class is a bit odd. There is a certain tendency to compress lines - e.g. to reach for one-liners, when in reality the body is not that trivial. If I were to write the code I'd insert a newline after the `{` and I'd also break apart initialization (e.g. no two statements separated by `;` in the same line).

Also, I noted that you start the body of the record (e.g. `{`) on a new line, which I also find odd.

Reply:
(to be clear, some of the above comments refer to the code that was already there before your changes)

Reply:
I also found the above as a bit odd but tried to stick with the existing implementation and style as much as possible.